File: ch21p746tableCellMenus2/ch21p718sections/MyCell.m


#import "MyCell.h"

@implementation MyCell

-(void)abbrev:(id)sender {
    NSLog(@"cell was told to abbrev %@", sender);
    if (self.celldelegate) {
        if ([self.celldelegate respondsToSelector:
             @selector(tableView:performAction:forRowAtIndexPath:withSender:)]) {
            // find my table view
            UIView* v = self;
            do {
                v = v.superview;
            } while (![v isKindOfClass:[UITableView class]]);
            UITableView* tv = (UITableView*) v;
            // ask it what index path we are
            NSIndexPath* ip = [tv indexPathForCell:self];
            // pretend to be the delegate
            [self.celldelegate tableView:tv performAction:_cmd forRowAtIndexPath:ip withSender:sender];
        }
    }
}


@end

File: iOS9bookExamples/bk2ch08p454tableCellMenus2/ch21p718sections/MyCell.swift


import UIKit

class MyCell : UITableViewCell {
    
    func abbrev(sender:AnyObject!) {
        // find my table view
        var v : UIView = self
        repeat {v = v.superview!} while !(v is UITableView)
        let tv = v as! UITableView
        // ask it what index path we are
        let ip = tv.indexPathForCell(self)!
        // talk to its delegate
        tv.delegate?.tableView?(tv, performAction:#selector(abbrev), forRowAtIndexPath:ip, withSender:sender)
    }

}
